### PR TITLE
show dishes on restaurant menu

### PR DESCRIPTION
**Pull request by sigu**
PT title: -> Add dishes to menu
PT Link: -> https://www.pivotaltracker.com/story/show/142697149

```
As a user
In order to decide from various dishes from a restaurant
I would like to see many dishes in the restaurant´s menu
```

### CODE REVIEW

File: spec/factories/dishes.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :dish do
+    name "MyString"
+    price "9.99"
```
Comment (tochman) [inline spec/factories/dishes.rb]: String?? Really?

File: app/controllers/menus_controller.rb [hunk 2]
```diff
@@ -2,6 +2,6 @@ class MenusController < ApplicationController
 
   def show
     @restaurant = Restaurant.find(params[:restaurant_id])
+    @dishes = @restaurant.menu.dishes
```
Comment (tochman) [inline app/controllers/menus_controller.rb]: Why store this in a separate variable if you can access dishes through the @restaurant object? Is there a point in having 2 variables? (I asked a similar question in the previous PR and would appreciate if you all followed in the discussion on code reviews)

File: spec/models/dish_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Dish, type: :model do
+  it 'has a valid factory' do
+    expect(FactoryGirl.create(:dish)).to be_valid
```
Comment (tochman) [inline spec/models/dish_spec.rb]: I thought you guys included the FactoryGirl methods so you could use the short syntax.

Comment (fottander) [reply spec/models/dish_spec.rb]: Yes of course. Fixed it.

File: app/controllers/menus_controller.rb (hunk 2 above)
Comment (fottander) [reply app/controllers/menus_controller.rb]: Now we use the restaurant variable instead.

File: features/add_dishes.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    | Tasty burgers   | AmberBurger |
+   And the following dishes exist
+    | name       | price | menu            |
+    | Kebabrulle | 180   | Nisses Takeaway |
+
+   Scenario: I see dishes in the menu
+    Given I am on the restaurant menu page for "Nisses Takeaway"
+    Then I should see "Menu"
+    And I should see "Kebabrulle"
+    And I should see "180"
```
Comment (tochman) [inline features/add_dishes.feature]: And what do you see for Tasty burgers?

**Comment by tochman** [comment]
I notice that you guys started with creating the dish model. That is good but I'd like to point out that the first step is always an acceptance test. Units should be created when they are needed. 😜

File: features/add_dishes.feature [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+   Given the following dishes exist
+    | name       | price | menu            |
+    | Kebabrulle | 180   | Nisses Takeaway |
+    | Burger     | 190   | Tasty burgers   |
+
+   Scenario: I see dishes in the menu
+    Given I am on the restaurant menu page for "Nisses Takeaway"
+    Then I should see "Kebabrulle"
+    And I should not see "Burger"
+    And I should see "180"
```
Comment (tochman) [inline features/add_dishes.feature:24]: Look, I still can't see any scenario for AmberBurger restaurant or the menu Tasty burgers.

File: features/add_dishes.feature [hunk 6]
```diff
@@ -7,14 +7,18 @@ Feature: Add dishes to menu
    Given the following restaurants exist
     | name        |
     | NisseKebbab |
-   And the following menus exist
+    | AmberBurger |
```
Comment (tochman) [inline features/add_dishes.feature]: Oh, so you removed the restaurant with an empty menu. Well, I'm still curious how the system reacts to an empty mnu, what do you display? A scenario for that could be in place.

File: features/add_dishes.feature (hunk 5 above)
Comment (fottander) [reply features/add_dishes.feature:24]: We are testing that in the menu_show_page.feature. To not see Amber Burger´s Tasty Burgers menu. 
Do we have to test that again here? Or am I not following?

Comment (tochman) [reply features/add_dishes.feature:24]: No, this is a good answer.

**Review by tochman — APPROVED** [review]

**Pull request merged**